routes/queryargs.go

```diff
@@ -0,0 +1,189 @@
+//   Copyright 2017 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package routes
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type (
+	// QueryArgInt is a type allowing to know the expected int
+	// type for the argument. QueryArgInt can be interfaced to
+	// QueryArgType.
+	QueryArgInt struct{}
+
+	// QueryArgUint is a type allowing to know the expected uint
+	// type for the argument. QueryArgUint can be interfaced to
+	// QueryArgType.
+	QueryArgUint struct{}
+
+	// QueryArgString is a type allowing to know the expected string
+	// type for the argument. QueryArgString can be interfaced to
+	// QueryArgType.
+	QueryArgString struct{}
+
+	// QueryArgIntSlice is a type allowing to know the expected []int
+	// type for the argument. QueryArgIntSlice can be interfaced to
+	// QueryArgType.
+	QueryArgIntSlice struct{}
+
+	// QueryArgUintSlice is a type allowing to know the expected []uint
+	// type for the argument. QueryArgUintSlice can be interfaced to
+	// QueryArgType.
+	QueryArgUintSlice struct{}
+
+	// QueryArgStringSlice is a type allowing to know the expected []string
+	// type for the argument. QueryArgStringSlice can be interfaced to
+	// QueryArgType.
+	QueryArgStringSlice struct{}
+
+	// QueryArgType is an interface used by all the type above. The Parse
+	// method takes the raw value of the argument and returns its typed value.
+	// An error can be returned if the value could not be parse. The error's
+	// message contains %s which has to be replaced by the argument's name
+	// before being displayed.
+	QueryArgType interface {
```

Comment:
Your godoc comments don’t need to include that `xxx is an interface/type/function`. We can see that very close to your documentation.